source/Plugins.cpp

```diff
@@ -0,0 +1,82 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	map<string, bool> settings;
+
+	void LoadSettings(const string &path)
+	{
+		DataFile prefs(path);
+		for(const DataNode &node : prefs)
+			settings[node.Token(0)] = (node.Size() == 1 || node.Value(1));
```

Comment:
Let's be just a bit more future-proof:
1. Store the data under a key, rather than treat the file itself as the key.
2. Require an argument, rather than treating the token alone as enabled.

Reply:
I used `preferences.txt` as a reference.  I think I understand what you mean but I don't have examples.  I'm currently using either `preferences.txt` or `keys.txt` as references and they're not under a key.

When you say key do you mean something like this:

```
plugins
	some-plugin 1
	another-plugin 0
```

I'm not sure what you mean by require an argument.  As in skip if there's not a token or value?

Reply:
> I used `preferences.txt` as a reference.  I think I understand what you mean but I don't have examples.  I'm currently using either `preferences.txt` or `keys.txt` as references and they're not under a key.
> 
> When you say key do you mean something like this:
> 
> ```
> plugins
> 	some-plugin 1
> 	another-plugin 0
> ```

Yep. Using a key will allow adding additional data that is not exactly of the same type/purpose to the file in a non-breaking manner while also letting us avoid order-dependent logic when we do that. I know it is highly desired for there to be pilot-specific plugin configurations, so while we don't hit that here, we should try to make getting there as simple as possible.

> I'm not sure what you mean by require an argument.  As in skip if there's not a token or value?

Yep, require each plugin line to have 2 tokens.

Reply:
Ok I understand